internal/gcpmanager/main.go

```diff
@@ -0,0 +1,108 @@
+// Package gcp provides a GCP implementation of the Substratus Cloud Interface (SCI).
+package gcpmanager
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	credentials "cloud.google.com/go/iam/credentials/apiv1"
+	storage "cloud.google.com/go/storage"
+	sci "github.com/substratusai/substratus/internal/sci"
+	credentialspb "google.golang.org/genproto/googleapis/iam/credentials/v1"
+)
+
+// server implements the sci.ControllerServer interface.
+type Server struct {
+	sci.UnimplementedControllerServer
+	StorageClient *storage.Client
+}
+
+// CreateSignedURL generates a signed URL for a specified GCS bucket and object path.
+func (s *Server) CreateSignedURL(ctx context.Context, req *sci.CreateSignedURLRequest) (*sci.CreateSignedURLResponse, error) {
+	bucketName, objectName := req.GetBucketName(), req.GetObjectName()
+
+	bucket := s.StorageClient.Bucket(bucketName)
+	obj := bucket.Object(objectName)
+	if _, err := obj.Attrs(ctx); err != nil {
+		if err != storage.ErrObjectNotExist {
+			// An error occurred that was NOT ErrObjectNotExist.
+			// This is an unexpected error and we should return it.
+			return nil, err
+		}
+		// the object doesn't exist and we want to continue and create the signed URL.
+	}
+
+	saEmail, err := getServiceAccountEmail()
```

Comment:
Currently this sends a call to the metadata server every time this RPC endpoint is invoked. This should be changed to do this once and store the service account email on the Server struct.

Reply:
I like that. Can do

Reply:
Done